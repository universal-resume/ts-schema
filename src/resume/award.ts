import { Array as ArraySchema, optional, Struct } from "effect/Schema";
import {
	Day,
	Location,
	Organization,
	ProperNoun,
	Summary,
	Tag,
} from "#value-object";

export namespace Award {
	export const Schema = Struct({
		title: ProperNoun.FromString,
		issuer: Organization.Schema,
		date: Day.FromString,
		location: optional(Location.Schema),
		summary: optional(Summary.FromString),
		tags: optional(ArraySchema(Tag.FromString)),
	});

	export type Type = typeof Schema.Type;
}
