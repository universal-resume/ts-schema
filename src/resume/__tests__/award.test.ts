import { Schema } from "effect";
import { expect, it } from "vitest";
import { Award } from "../award";

it("should parse this Award", () => {
	const input: unknown = {
		title: "IT Leadership Excellence Award",
		issuer: {
			name: "Parts Unlimited",
		},
		date: "2014",
		location: {
			address: "Phoenix, AZ",
			countryCode: "US",
		},
		summary:
			"Awarded for outstanding leadership during the digital transformation of Parts Unlimited's IT operations. Recognized for stabilizing critical systems and fostering DevOps adoption.",
		tags: ["Leadership", "IT Operations", "DevOps"],
	};
	const award = Schema.decodeUnknownSync(Award.Schema)(input);
	expect(award).toMatchSnapshot();
});
