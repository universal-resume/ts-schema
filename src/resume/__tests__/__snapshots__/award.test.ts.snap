// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should parse this Award 1`] = `
{
  "date": 2014-01-01T00:00:00.000Z,
  "issuer": {
    "name": "Parts Unlimited",
  },
  "location": {
    "address": "Phoenix, AZ",
    "countryCode": "US",
  },
  "summary": "Awarded for outstanding leadership during the digital transformation of Parts Unlimited's IT operations. Recognized for stabilizing critical systems and fostering DevOps adoption.",
  "tags": [
    "Leadership",
    "IT Operations",
    "DevOps",
  ],
  "title": "IT Leadership Excellence Award",
}
`;
